shell/platform/linux/fl_binary_messenger_test.cc

```diff
@@ -13,419 +13,367 @@
+  embedder_api->SendPlatformMessage = MOCK_ENGINE_PROC(
+      SendPlatformMessage,
+      ([&called](auto engine, const FlutterPlatformMessage* message) {
+        called = true;
 
-static gboolean send_ready_cb(gpointer user_data) {
-  FlFakeBinaryMessenger* self = FL_FAKE_BINARY_MESSENGER(user_data);
+        EXPECT_STREQ(message->channel, "test");
+        EXPECT_EQ(message->message_size, static_cast<size_t>(0));
 
-  self->send_callback(G_OBJECT(self), NULL, self->send_callback_user_data);
+        return kSuccess;
+      }));
 
-  return FALSE;
+  g_autoptr(FlBinaryMessenger) messenger = fl_binary_messenger_new(engine);
+  fl_binary_messenger_send_on_channel(messenger, "test", nullptr, nullptr,
+                                      nullptr, nullptr);
+  EXPECT_TRUE(called);
 }
 
-static void send_on_channel(FlBinaryMessenger* messenger,
-                            const gchar* channel,
-                            GBytes* message,
-                            GCancellable* cancellable,
-                            GAsyncReadyCallback callback,
-                            gpointer user_data) {
-  FlFakeBinaryMessenger* self = FL_FAKE_BINARY_MESSENGER(messenger);
-
-  EXPECT_STREQ(channel, "CHANNEL");
-  g_autofree gchar* text =
-      g_strndup(static_cast<const gchar*>(g_bytes_get_data(message, nullptr)),
-                g_bytes_get_size(message));
-  EXPECT_STREQ(text, "Marco!");
-
-  // Send response.
-  self->send_callback = callback;
-  self->send_callback_user_data = user_data;
-  g_idle_add(send_ready_cb, messenger);
-}
+// Checks sending a zero length message works.
+TEST(FlBinaryMessengerTest, SendEmpty) {
+  g_autoptr(FlDartProject) project = fl_dart_project_new();
+  g_autoptr(FlEngine) engine = fl_engine_new(project);
+  FlutterEngineProcTable* embedder_api = fl_engine_get_embedder_api(engine);
 
-static GBytes* send_on_channel_finish(FlBinaryMessenger* messenger,
-                                      GAsyncResult* result,
-                                      GError** error) {
-  const char* text = "Polo!";
-  return g_bytes_new(text, strlen(text));
-}
+  g_autoptr(GError) error = nullptr;
+  EXPECT_TRUE(fl_engine_start(engine, &error));
+  EXPECT_EQ(error, nullptr);
 
-static void resize_channel(FlBinaryMessenger* messenger,
-                           const gchar* channel,
-                           int64_t new_size) {
-  // Fake implementation. Do nothing.
-}
+  bool called = false;
+  embedder_api->SendPlatformMessage = MOCK_ENGINE_PROC(
+      SendPlatformMessage,
+      ([&called](auto engine, const FlutterPlatformMessage* message) {
+        called = true;
 
-static void set_warns_on_channel_overflow(FlBinaryMessenger* messenger,
-                                          const gchar* channel,
-                                          bool warns) {
-  // Fake implementation. Do nothing.
-}
+        EXPECT_STREQ(message->channel, "test");
+        EXPECT_EQ(message->message_size, static_cast<size_t>(0));
```

Comment:
Does this test have the same result as sending nullptr, or can the nullptr test also check that the data is indeed nullptr or something?

Reply:
I didn't test this as it is provided by the engine, and it's not defined by us if this is nullptr or pointing to some buffer. I checked and it is nullptr at least currently, but I've added a comment to note this.